src/aks-preview/azext_aks_preview/_validators.py

```diff
@@ -233,13 +233,12 @@ def validate_user(namespace):
         raise CLIError("--user can only be clusterUser or clusterMonitoringUser")
 
 
-def validate_vnet_subnet_id(namespace):
-    if namespace.vnet_subnet_id is not None:
-        if namespace.vnet_subnet_id == '':
-            return
-        from msrestazure.tools import is_valid_resource_id
-        if not is_valid_resource_id(namespace.vnet_subnet_id):
-            raise CLIError("--vnet-subnet-id is not a valid Azure resource ID.")
+def validate_subnet_id(namespace):
+    if namespace.vnet_subnet_id is None or namespace.vnet_subnet_id == '':
```

Comment:
miss to validate `namespace.pod_subnet_id` ?

Reply:
Thanks for catching that. Refactored validate subnet id to a helper with two different namespace functions that call it.